roles/rails_app/handlers/main.yml

```diff
@@ -1,7 +1,14 @@
+# To run the rails role without restarting services use --skip-tags restart_services
 ---
 - name: "restart rails app services"
   ansible.builtin.service:
     name: "{{ item }}"
     state: restarted
   loop: "{{ rails_app_site_config_services }}"
-  when: running_on_server
+  # the list of services includes `.service` on the end of each one, and you
+  # can't use template variables in conditionals, so we do this map stuff to
+  # remove the `.service` and then check for each service
+  when: >
+    running_on_server and
+    (item in services.keys()|list|map('split', '.')| map('first')) and
+    'restart_services' not in ansible_skip_tags
```

Comment:
This is an odd use of tags. I would have to remind myself why I need to skip a `restart_services` tag, when nothing in the role has that tag. Did you try marking the handler with `tags:` and adding both `restart_services` and `site_config`? That would be a more standard way to incorporate the same logic and use of tags when running a playbook or role.

Reply:
I didn't try it because my understanding was that you can't skip tags with a handler. But maybe that has changed -- I'll try it now!

Reply:
Yeah, it doesn't work :( 

Handlers have some kind of special logic so they kick in no matter what you do with tags. This `not in ansible_skip_tags` is a workaround for that.

Reply:
Gotcha. I'll keep thinking about alternatives, but meanwhile this solves our immediate problem.